### PR TITLE
Fix display of query string on pdo_error template.

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

File: src/Database/Exception/QueryException.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+{
+    protected LoggedQuery|string|null $query = null;
+
+    /**
+     * Set the query that caused this exception.
+     *
+     * @param \Cake\Database\Log\LoggedQuery|string $query
+     * @return void
+     */
+    public function setQuery(LoggedQuery|string $query): void
```
Comment (othercorey) [inline src/Database/Exception/QueryException.php]: Does it make sense for a QueryException to not have a Query set? Should we make it part of the constructor?

Comment (ADmad) [reply src/Database/Exception/QueryException.php]: Maybe change the constructor to `__construct(PDOException $exception, 
 LoggedQuery|string $query)`? The error message and code too can be fetched from PDOException itself.

File: src/Database/Exception/QueryException.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    /**
+     * Get the query string that caused this exception.
+     *
+     * @return string
+     */
+    public function getQueryString(): string
+    {
+        if ($this->query instanceof LoggedQuery) {
```
Comment (markstory) [inline src/Database/Exception/QueryException.php:42]: How does `query` get set on the exception?

Comment (ADmad) [reply src/Database/Exception/QueryException.php:42]: Through constructor property promotion :) Note the `protected` keyword for the `$query` argument for the constructor.

Comment (markstory) [reply src/Database/Exception/QueryException.php:42]: :blush: I totally missed that.

**Pull request merged**